src/MagnumPlugins/UfbxImporter/Test/UfbxImporterTest.cpp

```diff
@@ -357,7 +843,1651 @@ void UfbxImporterTest::light() {
+        {MaterialAttribute::BaseColor, Color4{0.02f, 0.03f, 0.04f, 1.0f} * factor(0.01f)},
+        {"diffuseRoughness", 0.05f},
+        {MaterialAttribute::Metalness, 0.06f},
+        {MaterialAttribute::SpecularColor, Color4{0.08f, 0.09f, 0.10f, 1.0f} * factor(0.07f)},
+        {MaterialAttribute::Roughness, 0.11f},
+        {"specularIor", 0.12f},
+        {"specularAnisotropy", 0.13f},
+        {"specularRotation", 0.14f},
+        {MaterialAttribute::EmissiveColor, Color3{0.52f, 0.5299f, 0.54f} * Vector3{0.51f}}, /* bad rounding in file */
+        {"thinFilmThickness", 0.55f},
+        {"thinFilmIor", 0.56f},
+        {"opacity", Color3{0.57f, 0.58f, 0.5899f}}, /* bad rounding in file */
+        {"indirectDiffuse", 0.63f},
+        {"indirectSpecular", 0.64f},
+
+        {MaterialAttribute::LayerName, "ClearCoat"},
+        {MaterialAttribute::LayerFactor, 0.35f},
+        {"color", Color4{0.36f, 0.37f, 0.38f, 1.0f}},
+        {MaterialAttribute::Roughness, 0.39f},
+        {"ior", 0.40f},
+        {"anisotropy", 0.41f},
+        {"rotation", 0.42f},
+
+        {MaterialAttribute::LayerName, "transmission"},
+        {MaterialAttribute::LayerFactor, 0.15f},
+        {"color", Vector4{0.16f, 0.17f, 0.18f, 1.0f}},
+        {"depth", 0.19f},
+        {"scatter", Vector3{0.20f, 0.21f, 0.22f}},
+        {"scatterAnisotropy", 0.23f},
+        {"dispersion", 0.24f},
+        {"extraRoughness", 0.25f},
+        {"enableInAov", true},
+        {"priority", Long(69)},
+
+        {MaterialAttribute::LayerName, "subsurface"},
+        {MaterialAttribute::LayerFactor, 0.26f},
+        {"color", Color4{0.27f, 0.28f, 0.29f, 1.0f}},
+        {"radius", Color3{0.30f, 0.31f, 0.32f}},
+        {"scale", 0.33f},
+        {"anisotropy", 0.34f},
+        {"type", Long(1)},
+
+        {MaterialAttribute::LayerName, "sheen"},
+        {MaterialAttribute::LayerFactor, 0.46f},
+        {"color", Color3{0.47f, 0.48f, 0.49f}},
+        {MaterialAttribute::Roughness, 0.50f},
+
+        {MaterialAttribute::LayerName, "matte"},
+        {MaterialAttribute::LayerFactor, 0.68f},
+        {"color", Color3{0.65f, 0.66f, 0.67f}},
+
+    }, {14, 21, 31, 38, 42, 45}};
+
+    CORRADE_COMPARE_AS(*material, reference, DebugTools::CompareMaterial);
+}
+
+void UfbxImporterTest::materialMayaArnoldFactors() {
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("UfbxImporter");
+    importer->configuration().setValue("preserveMaterialFactors", true);
+
+    CORRADE_VERIFY(importer->openFile(Utility::Path::join(UFBXIMPORTER_TEST_DIR, "maya-material-arnold.fbx")));
+    CORRADE_COMPARE(importer->materialCount(), 1);
+
+    Containers::Optional<MaterialData> material = importer->material("aiStandardSurface1");
+    CORRADE_VERIFY(material);
+
+    MaterialData reference{MaterialType::PbrMetallicRoughness|MaterialType::PbrClearCoat, {
+        {MaterialAttribute::BaseColor, Color4{0.02f, 0.03f, 0.04f, 1.0f}},
+        {"baseFactor", 0.01f},
```

Comment:
I was thinking about this, originally I had `baseColorFactor` but I felt `layerFactor` is a bit closer semantically to what it "represents", ie. the weight of the "base layer" lighting, to me it feels incidental that scaling the color has the same effect. Though I'm not invested in the name so if you prefer `layerColorFactor` I have no issue with it!

Reply:
It shouldn't affect the inner maps directly in any way, just the result of the shading. What I was trying to explain is that given the [OSL standard surface](https://autodesk.github.io/standard-surface/#toc3.8) definition:

```
base_mix = (1 - subsurface) * base * base_color * diffuse_brdf(...) + subsurface * subsurface_mix
```

This could be interpreted in two ways due to the properties of the multiplication. Here I would call the first one `baseColorFactor` and the second one `baseFactor`, the end result is equivalent anyways.

```
base_mix = (1 - subsurface) * (base * base_color) * diffuse_brdf(...) + subsurface * subsurface_mix
base_mix = (1 - subsurface) * base * (base_color * diffuse_brdf(...)) + subsurface * subsurface_mix
```

It's extra confusing as the OSL whitepaper actually uses *both* interpretations in different points for **base**:

![image](https://user-images.githubusercontent.com/3875461/207148375-a869c2cd-e2c8-4a66-9987-987c6feb00e3.png)

![image](https://user-images.githubusercontent.com/3875461/207148305-3f17c031-ca42-4310-b49f-f355cc9db7ff.png)

So all in all I don't mind which one we go with, both feel reasonably justified for me (I just prefer `baseFactor` for some reason, maybe because I have been staring at `base` and `base_weight` for so long in FBX material definitions :D). But really I have no objection with `baseColorFactor` either, at least that's straightforward with how to apply it when shading.